session/setup_test.go

```diff
@@ -0,0 +1,81 @@
+package session
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/alicebob/miniredis"
+	"github.com/garyburd/redigo/redis"
+	"github.com/superfly/tlstest"
+)
+
+var (
+	redisPool       *redis.Pool
+	serverTLSConfig *tls.Config
+	clientTLSConfig *tls.Config
+
+	serverTLSCert tls.Certificate
+	serverCrtPEM  []byte
+	serverKeyPEM  []byte
+
+	testRedis *miniredis.Miniredis
+)
+
+// One place to define setup and teardown for all tests in session package
+func TestMain(m *testing.M) {
```

Comment:
Ah nice one on this! I like us pulling this out into a more central, non hidden location